kitsune/questions/static/questions/js/components/AAQApp.jsx

```diff
@@ -39,17 +40,30 @@ export default class AAQApp extends React.Component {
       validationErrors: QuestionEditStore.getValidationErrors(),
       questionState: QuestionEditStore.getState(),
       userAuth: UserAuthStore.getAll(),
-      troubleshooting: TroubleshootingDataStore.getAll(),
+      troubleshooting: TroubleshootingDataStore.getAll()
     };
   }
 
+  setStep(step) {
+    this.setState({step: step});
+  }
+
   render() {
     return (
       <div className="AAQApp">
-        <ProductSelector {...this.state}/>
-        <TopicSelector {...this.state}/>
-        <TitleContentEditor {...this.state}/>
-        <UserAuth userAuth={this.state.userAuth}/>
+        {(() => {
+          switch (this.state.step) {
+            case 'topic':
+              return <TopicSelector {...this.state} setStep={this.setStep.bind(this)} next="title"/>;
+            case 'title':
+              return [
+                <TitleContentEditor {...this.state}/>,
+                <UserAuth userAuth={this.state.userAuth}/>
+              ];
+            default:
+              return <ProductSelector {...this.state} setStep={this.setStep.bind(this)} next="topic"/>;
+          }
+        })()}
```

Comment:
Wow. That's a lot of parenthesis. I've never seen a iife put into a JSX context before, but it works pretty well.

I think I probably would have put this outside the JSX like below, but this works fine too. I wouldn't hvae thought to do it this way

```javascript
render() {
  let step;
  switch (this.state.step) {
    case 'topic':
      step = ...;    
    ...
  }

  return (
    <div className="AAQApp">
      <step/>
      <SubmitQuestion {...this.state}/>
    </div>
  }
```